docs/src/main/asciidoc/central-logging.adoc

```diff
@@ -0,0 +1,377 @@
+
+== Example application
+
+The following examples will all be based on the same example application that you can create based on the following steps:
+
+- Create an application with the `quarkus-logging-gelf` extension. You can use the following Maven command to create it:
+```
+mvn io.quarkus:quarkus-maven-plugin:1.0.0.Final:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=gelf-logging \
+    -DclassName="org.acme.quickstart.GelfLoggingResource" \
+    -Dpath="/gelf-logging" \
+    -Dextensions="logging-gelf"
+```
+
+- Add a log inside your rest endpoint
+```
+@Path("/gelf-logging")
+@ApplicationScoped
+public class GelfLoggingResource {
+    private static final Logger LOG = Logger.getLogger(GelfLogHandlerResource.class);
+
+    @GET
+    public void log() {
+        LOG.info("Some useful log message");
+    }
+
+}
+```
+
+- Configure the gelf log handler to send logs to an external UDP endpoint on the port 12201
+```
+quarkus.log.handler.gelf.enabled=true
+quarkus.log.handler.gelf.host=localhost
+quarkus.log.handler.gelf.port=12201
+```
+
+WARNING: UDP logging is currently not supported on native mode, use TCP if you plan to deploy your application as a native image.
+
+== Send logs to Graylog
+
+To send logs to Graylog, you first need to launch the components that compose the Graylog stack:
+
+- MongoDB
+- Elasticsearch
+- Graylog
+
+You can do this via the following docker-compose file that you can launch via `docker-compose run -d`:
+
+```
+version: '3.2'
+
+services:
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch-oss:6.8.2
+    ports:
+      - "9200:9200"
+    environment:
+      ES_JAVA_OPTS: "-Xms512m -Xmx512m"
+    networks:
+      - graylog
+
+  mongo:
+    image: mongo:4.0
+    networks:
+      - graylog
+
+  graylog:
+    image: graylog/graylog:3.1
+    ports:
+      - "9000:9000"
+      - "12201:12201/udp"
+      - "1514:1514"
+    environment:
+      GRAYLOG_HTTP_EXTERNAL_URI: "http://127.0.0.1:9000/"
+    networks:
+      - graylog
+    depends_on:
+      - elasticsearch
+      - mongo
+
+networks:
+  graylog:
+    driver: bridge
+```
+
+Then, you need to create a UDP input in Graylog.
+You can do it from the Graylog web console (System -> Input -> Select GELF UDP) or via the API.
+
+This curl example will create a new Input of type GELF UDP, it uses the default login from Graylog (admin/admin).
+
+```
+curl -H "Content-Type: application/json" -H "Authorization: Basic YWRtaW46YWRtaW4=" -H "X-Requested-By: curl" -X POST -v -d \
+'{"title":"udp input","configuration":{"recv_buffer_size":262144,"bind_address":"0.0.0.0","port":12201,"decompress_size_limit":8388608},"type":"org.graylog2.inputs.gelf.udp.GELFUDPInput","global":true}' \
+http://localhost:9000/api/system/inputs
+```
+
+Launch your application, you should see your logs arriving inside Graylog.
+
+NOTE: due to the initialization time of the GELF log handler, it is possible that not all startup logs are sent to Graylog.
```

Comment:
Because of the way we set up our log handlers, the few "missed" log messages should be automatically replayed.  If this isn't working, let's have a chat to figure out the cause of the problem.

Reply:
Yes, it's the case. I wasn't sure so I write this note. Will delete it.

Reply:
deleted